pkg/apis/tenancy/v1alpha1/helper/helper.go

```diff
@@ -55,13 +55,30 @@ func EncodeOrganizationAndClusterWorkspace(organization, workspace string) strin
 	return organization + separator + workspace
 }
 
+// HasClusterWorkspace indicates whether provided org and workspace
+// identify a cluster that is backed by a ClusterWorkspace resource.
+func HasClusterWorkspace(org, ws string) bool {
+	return !(
+	// Root cluster
+	org == "" && ws == RootCluster ||
+		// Shard-local system cluster
+		strings.HasPrefix(org, LocalSystemClusterPrefix))
+}
+
 // WorkspaceKey returns a key to use when looking up a ClusterWorkspace in a lister or indexer.
 // If org is the value of OrganizationCluster, the key will be of the format
 // <OrganizationCluster>#$#<ws>. Otherwise, the key will be of the format
 // <OrganizationClsuter>_<org>#$#<ws>.
-func WorkspaceKey(org, ws string) string {
-	if org == RootCluster || (org == "" && ws == RootCluster) || strings.HasPrefix(org, LocalSystemClusterPrefix) {
-		return clusters.ToClusterAwareKey(org, ws)
+func WorkspaceKey(org string, ws string) string {
+	if org == "" && ws == RootCluster {
+		panic("The root cluster is not backed by a ClusterWorkspace")
+	}
+	if strings.HasPrefix(org, LocalSystemClusterPrefix) {
```

Comment:
would go one step further: check for colon. Colon is a sign that we did something wrong.

Reply:
+1